service.py

```diff
@@ -107,11 +115,11 @@ def get_contact(row):
         for contact_tuple in phone_email_etc_list
         if len(contact_tuple) > 1
     }
-    phone = phone_email_etc_hash["Phone"]
+    phone = phone_email_etc_hash["P"] or phone_email_etc_hash["Phone"]
```

Comment:
I believe this will still error if "P" isn't there. I think you'd want to do something like
```
phone = phone_email_etc_hash.get("P", phone_email_etc_hash["Phone"])
```

Reply:
Looks like it errors if the backup value isn't present with the hash.get method; resorting to a try/except pattern

Reply:
If the backup isn't there, that means both are missing, and it should error. But something like this works:
```python
h={"a":1}
h["a"]
1
h.get("b")
h.get("b",h["a"])
1
```

Reply:
sounds good